tests/rename/24.t

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+# vim: filetype=sh noexpandtab ts=8 sw=8
+
+desc="rename of a directory updates its .. link"
+
+dir=`dirname $0`
+. ${dir}/../misc.sh
+
+echo "1..13"
+
+src_parent=`namegen`
+dst_parent=`namegen`
+src=`namegen`
+dst=`namegen`
+
+expect 0 mkdir ${src_parent} 0755
+expect 0 mkdir ${dst_parent} 0755
+expect 0 mkdir ${src_parent}/${src} 0755
+cdir=`pwd`
+
+# Initial conditions
+expect 3 lstat ${src_parent} nlink
```

Comment:
One of the advantages of #56 is that we can mark this test as requiring nlink, which object file systems like gcsfuse and s3fs do not/cannot efficiently support.